contrib/babelfishpg_tsql/sql/sys_function_helpers.sql

```diff
@@ -9966,17 +9966,17 @@ BEGIN
 		v_integral_digits := v_digits;
 	END IF;
 	IF (v_style = 0) THEN
-		v_format := (pow(10, v_integral_digits)-1)::TEXT || 'D99';
-		v_result := to_char(v_moneyval, v_format);
+		v_format := (pow(10, v_integral_digits)-10)::TEXT || 'D99';
+		v_result := pg_catalog.btrim(to_char(v_moneyval, v_format));
 	ELSIF (v_style = 1) THEN
-		IF (v_moneysign::SMALLINT = 1) THEN
-			v_result := substring(p_moneyval::TEXT, 2);
+		IF (v_moneysign::SMALLINT = -1) THEN
+			v_result := substring(p_moneyval::PG_CATALOG.MONEY::TEXT, 1, 1) || substring(p_moneyval::PG_CATALOG.MONEY::TEXT, 3);
 		ELSE
-			v_result := substring(p_moneyval::TEXT, 1, 1) || substring(p_moneyval::TEXT, 3);
+			v_result := substring(p_moneyval::PG_CATALOG.MONEY::TEXT, 2);
```

Comment:
Don't we require `btrim` here for style 1?

Reply:
No, in style 0, 2 and 126 we are using `to_char()` function which prepends space in the result, to trim that we used btrim there. But for style 1 we are using substring function so we do not need btrim.